amd_openvx_extensions/amd_media/encoder.cpp

```diff
@@ -331,57 +338,32 @@ vx_status CLoomIoMediaEncoder::Initialize()
-    cl_context context = nullptr;
-    ERROR_CHECK_STATUS(vxQueryContext(vxGetContext((vx_reference)node), VX_CONTEXT_ATTRIBUTE_AMD_OPENCL_CONTEXT, &context, sizeof(context)));
-    cl_device_id device_id = nullptr;
-    ERROR_CHECK_STATUS(clGetContextInfo(context, CL_CONTEXT_DEVICES, sizeof(device_id), &device_id, nullptr));
-#if defined(CL_VERSION_2_0)
-    cmdq = clCreateCommandQueueWithProperties(context, device_id, 0, nullptr);
-#else
-    cmdq = clCreateCommandQueue(context, device_id, 0, nullptr);
-#endif
-    ERROR_CHECK_NULLPTR(cmdq);
-    for (int i = 0; i < ENCODE_BUFFER_POOL_SIZE; i++) {
-        mem[i] = clCreateBuffer(context, CL_MEM_READ_WRITE, offset + stride * height, nullptr, nullptr);
-        ERROR_CHECK_NULLPTR(mem[i]);
-    }
-
-#elif ENABLE_HIP
-    int hip_device = -1;
-    ERROR_CHECK_STATUS(vxQueryContext(vxGetContext((vx_reference)node), VX_CONTEXT_ATTRIBUTE_AMD_HIP_DEVICE, &hip_device, sizeof(hip_device)));
-    if (hip_device < 0) {
-        return VX_FAILURE;
-    }
-    hipError_t err;
-
-    for (int i = 0; i < ENCODE_BUFFER_POOL_SIZE; i++) {
-        err = hipHostMalloc((void **)&hostBuffer[i], offset + stride * height);
-        if (err != hipSuccess) {
-            vxAddLogEntry(NULL, VX_FAILURE, "ERROR: hipHostMalloc => %d (failed)\n", err);
+    if (m_enableUserBufferGPU) {
+    #if ENABLE_OPENCL
+        cl_context context = nullptr;
+        ERROR_CHECK_STATUS(vxQueryContext(vxGetContext((vx_reference)node), VX_CONTEXT_ATTRIBUTE_AMD_OPENCL_CONTEXT, &context, sizeof(context)));
+        cl_device_id device_id = nullptr;
+        ERROR_CHECK_STATUS(clGetContextInfo(context, CL_CONTEXT_DEVICES, sizeof(device_id), &device_id, nullptr));
+      #if defined(CL_VERSION_2_0)
+        cmdq = clCreateCommandQueueWithProperties(context, device_id, 0, nullptr);
+      #else
+        cmdq = clCreateCommandQueue(context, device_id, 0, nullptr);
+      #endif
+        ERROR_CHECK_NULLPTR(cmdq);
+    #elif ENABLE_HIP
+        int hip_device = -1;
+        ERROR_CHECK_STATUS(vxQueryContext(vxGetContext((vx_reference)node), VX_CONTEXT_ATTRIBUTE_AMD_HIP_DEVICE, &hip_device, sizeof(hip_device)));
+        if (hip_device < 0) {
             return VX_FAILURE;
         }
-        if (hip_dev_prop.canMapHostMemory) {
-            err = hipHostGetDevicePointer((void **)&mem[i], hostBuffer[i], 0 );
-            if (err != hipSuccess) {
-                vxAddLogEntry(NULL, VX_FAILURE, "ERROR: hipHostGetDevicePointer => %d (failed)\n", err);
-                return VX_FAILURE;
-            }
-        } else {
-            err = hipMalloc(&mem[i], offset + stride * height);
-            if (err != hipSuccess) {
-                vxAddLogEntry(NULL, VX_FAILURE, "ERROR: hipMalloc(%p) => %d (failed)\n", mem[i], err);
-                return VX_FAILURE;
-            }
+        int bufHeight = (inputFormat == AV_PIX_FMT_NV12) ? (height + (height>>1)) : height;
+        hostBuffer = (uint8_t *)malloc(offset + stride * bufHeight);
```

Comment:
please change to hipHostMalloc to create pinned host memory.

Reply:
done